### PR TITLE
RE-1558 Add Phobos Region to clouds.yaml

### PR DESCRIPTION
**Pull request by dealako**
- Added phobos-nodepool profile to clouds.yaml
- Added additional credentials/env vars to cloud_creds to support phobos nodepool
- Refactored writeCloudCfg() routine and calls - moved outer cloud credentials wrapper to inside the writeCloudCfg() method where it is directly used.
- set fail_on_errors: False to ensure Ansible script doesn't just fail even if one VPN connection fails

JIRA: RE-1558

Issue: [RE-1558](https://rpc-openstack.atlassian.net/browse/RE-1558)

### CODE REVIEW

File: pipeline_steps/common.groovy [hunk 1]
```diff
@@ -1074,7 +1087,13 @@ List build_creds_array(String list_of_cred_ids){
     Map creds_bundles = [
       "cloud_creds": ['dev_pubcloud_username',
                       'dev_pubcloud_api_key',
-                      'dev_pubcloud_tenant_id'],
+                      'dev_pubcloud_tenant_id',
+                      'phobos_nodepool_auth_url',
```
Comment (odyssey4me) [inline pipeline_steps/common.groovy]: ``Could not find credentials entry with ID 'phobos_nodepool_auth_url'`` is the jenkins failure, indicating that either this is the wrong name or it's not in the credentials store.

**Comment by dealako** [comment]
recheck

**Comment by dealako** [comment]
recheck_unit_tests

File: pipeline_steps/pubcloud.groovy [hunk 2]
```diff
@@ -92,27 +82,22 @@ String getPubCloudSlave(Map args){
       if (!("inventory" in args)){
         args.inventory = "inventory"
       }
-      withCredentials(common.get_cloud_creds()){
```
Comment (mattt416) [inline pipeline_steps/pubcloud.groovy]: Is it the removal of this that's causing https://rpc.jenkins.cit.rackspace.net/job/RE-unit-test-phobos-vpn/992/ to fail?

Comment (dealako) [reply pipeline_steps/pubcloud.groovy]: wrapped with `withRequestedCredentials("cloud_creds") {}` block.

Comment (dealako) [reply pipeline_steps/pubcloud.groovy]: recheck_unit_tests

**Comment by mancdaz** [comment]
recheck_unit_tests

**Comment by mancdaz** [comment]
https://rpc.jenkins.cit.rackspace.net/job/RE-unit-test-phobos-vpn/1016/console


```
Caught exception on rpc-jenkins-n02.prod.dfw1.cit.rackspace.net-9082c1d1:
org.jenkinsci.plugins.credentialsbinding.impl.CredentialNotFoundException:
Could not find credentials entry with ID 'service_account_jenkins_api_username'
```

**Comment by mattt416** [comment]
💥

**Pull request merged**